Xamarin.Forms.Platform.UAP/LabelRenderer.cs

```diff
@@ -152,6 +158,26 @@ protected override void OnElementPropertyChanged(object sender, PropertyChangedE
 			base.OnElementPropertyChanged(sender, e);
 		}
 
+		void UpdateTextDecorations()
+		{
+			if (!Element.IsSet(Label.TextDecorationsProperty))
+				return;
+
+			var elementTextDecorations = Element.TextDecorations;
+
+			if ((elementTextDecorations & TextDecorations.Underline) == 0)
+				Control.TextDecorations &= ~Windows.UI.Text.TextDecorations.Underline;
+			else
+				Control.TextDecorations |= Windows.UI.Text.TextDecorations.Underline;
+
+			if ((elementTextDecorations & TextDecorations.Strikethrough) == 0)
+				Control.TextDecorations &= ~Windows.UI.Text.TextDecorations.Strikethrough;
+			else
+				Control.TextDecorations |= Windows.UI.Text.TextDecorations.Strikethrough;
+
+			Control.Text = Control.Text; //TextDecorations are not updated in the UI until the text changes
```

Comment:
This was an oddity I observed on UWP -- it seems that changes to `TextDecorations` do not propagate to the UI until the next time the text changes, as if `TextDecorations` only applies to each *new* string assigned to `Text`.

Reply:
So I think this is the code that is causing the exception on UWP.  If the Control TEXT is set via "spans" then this line of code wipes out the inline runs on the control which then causes the exception to happen in *RecalculateSpanPositions* so this code here is going to need to check if text is set via spans and if so then iterate over those instead of setting the Text to itsefl

Reply:
So I think this is the code that is causing the exception on UWP.  If the Control TEXT is set via "spans" then this line of code wipes out the inline runs on the control which then causes the exception to happen in *RecalculateSpanPositions* so this code here is going to need to check if text is set via spans and if so then iterate over those instead of setting the Text to itsefl